test/npmignore.js

```diff
@@ -0,0 +1,69 @@
+import path from 'path';
+import test from 'ava';
+import mockery from 'mockery';
+import sinon from 'sinon';
+
+let testedModule;
+
+const gitUtilApi = {
+	newFilesSinceLastRelease: async () => {}
+};
+
+const pkgDirApi = {
+	sync: () => {}
+};
+
+const stubPkgDir = sinon.stub(pkgDirApi, 'sync');
+
+test.before(() => {
+	const stubGitUtil = sinon.stub(gitUtilApi, 'newFilesSinceLastRelease');
+
+	mockery.registerAllowable('../source/util');
+	mockery.registerAllowable('../source/npm/util');
+	mockery.registerMock('./git-util', gitUtilApi);
+	mockery.registerMock('pkg-dir', pkgDirApi);
+
+	stubGitUtil.returns(['source/ignore.txt', 'source/pay_attention.txt', '.hg', 'test/file.txt']);
+
+	mockery.enable({
+		useCleanCache: true,
+		warnOnReplace: false,
+		warnOnUnregistered: false
+	});
+
+	testedModule = require('../source/util');
+});
+
+test.after(() => {
+	mockery.deregisterAll();
+	mockery.disable();
+});
+
+test.serial('ignored files using file-attribute in package.json with one file', async t => {
```

Comment:
Why do these tests need to be run serially?

Reply:
The result of `stubPkgDir` is set in every test-method. When the tests run parallel, the `stubPkgDir` in the first test-method was overwritten by the second test-method ....

Reply:
In that case ,I'd prefer it if we defined `stubPkgDir` inside each test that uses it, to make the tests run faster and make sure they are as encapsulated and self-contained as possible.

Reply:
Done. According to (https://github.com/avajs/ava/issues/1066) there seems to be no other solution, than run the test serially.

Reply:
@sindresorhus Do you think the new solution is better, or would it be better to run the tests serially (and avoid the code duplication)?

Reply:
@itaisteinherz One possible solution to avoid `test.serial` would be using `proxyquire`. But proxyquire can't mock or stub nested modules. As a workaround i could build a chain of mocked or stubbed modules with proxyquire.

Reply:
No better way when you stub. That's one of the reasons I prefer real tests using temporary directories rather than stubbing.

Reply:
@sindresorhus 
I moved to test `source/npm/util` as `source/util` doesn't have much logic inside. The `fixtures`-folder contains the temporary directories. Stubbing is just needed to swing the package directory to these temporary directories.